src/liberty/libertyProject.ts

```diff
@@ -173,25 +196,48 @@ export class ProjectProvider implements vscode.TreeDataProvider<LibertyProject>
 		for (const gradleBuild of validGradleBuilds) {
 			// if a LibertyProject for this build.gradle has already been created
 			// we want to re-use it
-			if (this.projects.has(gradleBuild.getBuildFilePath())) {
-				// check version of liberty-gradle-plugin to see if it is valid for contatiners
-				const project = this.projects.get(gradleBuild.getBuildFilePath());
-				if (project !== undefined) {
-					if (project.contextValue != gradleBuild.getProjectType()) {
-						project.setContextValue(gradleBuild.getProjectType());
-					}
-				}
-
-				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-				newProjectsMap.set(gradleBuild.getBuildFilePath(), this.projects.get(gradleBuild.getBuildFilePath())!);
-			}
-			// else we create a new LibertyProject for that build file
-			else {
+			if ( !this.addExistingProjectToNewProjectsMap(gradleBuild.getBuildFilePath(), gradleBuild.getProjectType(),
+					newProjectsMap) ) {
 				const project = await createProject(this._context, gradleBuild.getBuildFilePath(), gradleBuild.getProjectType());
 				newProjectsMap.set(gradleBuild.getBuildFilePath(), project);
 			}
 		}
 
+		const wsFolders = vscode.workspace.workspaceFolders;
+		if ( wsFolders ) {
+			for ( const folder of wsFolders ) {
+				const path = folder.uri.fsPath;
+				if ( this.projectRootPathExists(path, newProjectsMap.keys() ) === false ) {
+					const includeRelativePath = new vscode.RelativePattern(path, '**/src/main/liberty/config/server.xml')
+					const paths = (await vscode.workspace.findFiles(includeRelativePath, EXCLUDED_DIR_PATTERN)).map(uri => uri.fsPath);
+					serverXMLPaths = serverXMLPaths.concat(paths);
+				}
+			}
+		}
+
+		/* 
+		 * Finde the projects by server.xml
+		 */
+		for (const serverXML of serverXMLPaths) {
+			const folder = vscodePath.parse(vscodePath.resolve(serverXML, '../../../../../')).dir;
```

Comment:
Is this assuming that the server.xml is always in the "src/main/liberty/config" dir and the build file is at the root level of the project?

Reply:
yes, same as eclipse plugin.

Reply:
Consider a comment as well to explain the relative path resolution.

Note that this logic may break the multi module case, where we would want to add the root level pom.xml rather than the module level. Could be changed here or as part of #98 when official support for the multi-module Maven projects are added.

Reply:
Will add a comment to explain that. Thanks